### PR TITLE
File upload improvements

### PR DESCRIPTION
**Pull request by tanthammar**
File upload fixes:

**If original file name is not preserved**
- fix: prevents Livewire temporary file names longer than 255 chars, uses Str::ulid() instead.

**If original file name is preserved**
- new: explanatory file name with incrementing versioning
   - example: `file-foo.png` becomes `file-foo-edit-1.png`, `file-foo-edit-2.png`
   - covers previously saved file with custom naming, containing a version,
       -  like `file-foo-edit-1-suffix.png` becomes `file-foo-edit-2-suffix.png`

**Other**
- fix: catches filepond add file error, previously hidden behind the editor modal.
- new: confirmation to warn when editing svg files which completely destroys the image quality.
- new: method to disable/enable editing of svg files, disabled by default
- fix: #7703 
- fix: filepond upload button sometimes hidden when editing multiple times

**Example of cropped svg file**

Before:
image

After:
image

### CODE REVIEW

**Comment by tanthammar** [comment]
There are still CSS issues with multiple file uploads when cropping multiple times. The button disappears.
Will work on that tomorrow.

**Comment by tanthammar** [comment]
> There are still CSS issues with multiple file uploads when cropping multiple times. The button disappears. Will work on that tomorrow.

done

**Comment by wychoong** [comment]
imo filename length 255 chars should be fixed in LW, as it will encode it and will failed in LW before reaching filament 

see https://github.com/filamentphp/filament/issues/8841#issuecomment-1751199793

**Comment by wychoong** [comment]
Not too sure about the confirmation prompt for svg. Can the quality issue be solved by converting the svg into image before editing?

https://github.com/fengyuanchen/cropperjs/issues/733
https://github.com/pqina/filepond-plugin-image-preview/issues/5

**Comment by tanthammar** [comment]
@wychoong My impression is that the link you provided for the svg issues are unrelated. One is a about a Firefox bug and the other is resolved. 
FilePond works very nice with svgs.
It is the html canvas to blob that is destructive.

Do you have another solution for svgs?

**Comment by wychoong** [comment]
I’ll will try pull it down to see if there is another way

**Comment by wychoong** [comment]
Attic open livestream

**Comment by wychoong** [comment]
couldnt replicate the destructive svg you facing

I'm using latest filament and LW
tested with this svg file
map

in the editor I tried 
- cropping without zooming
- zoom in without resize cropper
booth scenario the file saved is in good quality
tested on Edge,Firefox,Safari

saved file
c1RZ7qH6AruCdtXeDD72Vbec0HiUCM-metabWFwLnN2Zw==-

**Comment by wychoong** [comment]
The issue I found when testing is on Create form, the file edited is not saved on the record, but both original and edited files are stored on filesystem, separate issue that I might open an issue later

**Comment by tanthammar** [comment]
Thank you @wychoong , will look into it.

**Comment by tanthammar** [comment]
**Changes**

1. SVG editing is enabled by default.
2. Maintain `mb-0` on the `filepond--root` element.
3. Remove the alert on input errors that are hidden behind the modal. Simply close the modal. The error will be displayed by Filepond or as a regular validation message.
4. Incremental version naming has changed from `edit-#` to `v#`.
    - Example: `file-foo-v1.png`, `file-foo-v2.png`, and so on.
    - Custom naming with suffixes is supported, such as `file-foo-v#-suffix.png`.


**Regarding SVG Files**
After conducting investigations, testing, and reviewing the source code of multiple SVG files, I have concluded that an SVG file must have `width` and `height` attributes with defined units in points to achieve an acceptable quality SVG-to-PNG conversion.

**Criteria that must be met:**
- An SVG file must include width and height attributes.
- The width and height must have defined units.
- For optimal results, the units should be in pt (points).
- The width and height must match the values in the viewBox attribute (which they often don't).

If these criteria are not met, the conversion can be highly destructive.

**Solution**
Process all SVG files before passing them to the image editor by adding or updating the width and height in points. Utilize the viewBox attribute to set the dimensions. (Note that viewBox may have different casings; this PR includes case-insensitive matching.)

**SVG confirmation and disabled alert**
I would like to retain this features, even if it means creating more translations.

**Scenario**
Consider a user who uploads an image in November. Next summer, they want to crop it. The Filepond input field doesn't display the full filename, making it impossible to discern the file extension. Furthermore, many of my users are not well-versed in the distinctions between various file formats. Without a warning before cropping the image, the harm is done.

Even if this PR optimizes SVGs for the best possible resolution, the conversion remains destructive. The infinite scalability is lost.

**Benefits of Keeping the Feature**

- **Prevent Unintentional Conversions:** By offering an option to disable SVG editing or displaying a confirmation warning, we empower users to make informed decisions. This prevents unintentional conversions and safeguards the valuable features of SVG files.
- **Enhanced User Experience:** The solution enhances the user experience by providing transparency and control over the editing process. Users can confidently edit their images without the fear of accidental data loss. Additionally, the alert displayed when editing of SVG files is disabled is necessary to inform the user why nothing happens when they click the editor button.

**Comment by wychoong** [comment]
Question
- can the alert/confirmation be disabled
- If the viewbox from svg doesn’t fulfill your checking I think it will still load into editor and ends up destructive? Is it possible to render with a fixed width and get the height to “patch” the svg

**Comment by tanthammar** [comment]
Changes:
Both the alert and the confirmation are disabled by default.

**Comment by tanthammar** [comment]
> Question
> 
> * can the alert/confirmation be disabled
> * If the viewbox from svg doesn’t fulfill your checking I think it will still load into editor and ends up destructive? Is it possible to render with a fixed width and get the height to “patch” the svg

@wychoong 
I have yet to find any SVG file without the viewBox attribute. I only found that the casing could be different.
Don't really know which width/height to use as a fallback, because it would stretch/distort the image if the ratio is incorrect.
Do you have a suggestion?


(The alert/confirmation are disabled now by default.)

File: packages/forms/resources/css/components/file-upload.css [hunk 1]
```diff
@@ -5,7 +5,7 @@
 @import 'filepond-plugin-media-preview/dist/filepond-plugin-media-preview.css';
 
 .filepond--root {
-    @apply mb-0 overflow-hidden;
+    margin-bottom: 0;
```
Comment (danharrin) [inline packages/forms/resources/css/components/file-upload.css]:
```suggestion
    @apply mb-0;
```

**Comment by sinnbeck** [comment]
It seems that this PR might have made some unwanted changes to the upload field that makes it extend below, overlaying other elements

image

This means it isn't possible to click the button anymore

**Comment by danharrin** [comment]
Unfortunately, the overflow-hidden was also causing issues too, can you try to find a different solution rather than adding it back?

**Comment by sinnbeck** [comment]
I will see if I can find a way to get it all working but I am just on my way on vacation so it might be a while sadly

**Comment by sinnbeck** [comment]
It seems that filepond adds a translate3d that drops it below the input. Think a source dive in filepond is needed.

image

**Comment by sinnbeck** [comment]
Adding this css style as an override seems to fix it, but it might be worth diving deeper into the filepond code. I will dig into it more when I get back

```css
.filepond--list-scroller {
    top: -100% !important;
}

**Comment by sinnbeck** [comment]
In case anyone else has the same issue. It seems that the admin panel works with this (it still overflows but the buttons work) so I dug into why. It is all down to if the button has the class `relative`. So just add that to your buttons/element and it should work

**Comment by danharrin** [comment]
Is it worth PRing the fix for `top`?

**Comment by sinnbeck** [comment]
I had to remove that again as it caused other problems sadly. Ensuring overlapping element has positon: relative seems to be the best way of fixing it

**Comment by illia-sapryga** [comment]
I am having same issue

**Comment by illia-sapryga** [comment]
@danharrin can you please look into this issue ? I tried many thing and it breaks something in all cases.

**Comment by danharrin** [comment]
I do not fix issues without reproduction repositories. Also, it sounds like a Filepond issue rather than a Filament issue.

**Pull request merged**